app/controllers/incurred_incidentals_controller.rb

```diff
@@ -56,6 +57,20 @@ def update
 
   private
 
+  def upload_documents
+    # only do this on sucess and with a file
+    if @incurred_incidental.errors.empty? && params[:file]
+      # only allow types of uploaded file
+      params.require(:file).transform_values! { |uf| uf if uf.class == ActionDispatch::Http::UploadedFile }
+
+      params.require(:file).to_unsafe_h.map(&:itself).each do |id, uploaded_file|
```

Comment:
i don't remember finding one and I agree, I would rather find a safer way to do it than simply allowing everything.   This line here does something super simple, it takes a hash like  `{ "1" => UploadedFile, "2" => UploadedFile }` and transforms it into `[["1", UploadedFile],["2", UploadedFile]]` then the each block takes the parts uses them to store the file.